stdlib/domain.ml

```diff
@@ -145,19 +145,20 @@ module DLS = struct
     | Some e -> set_dls_list (e::vals)
 
   let get k =
-    let vals = get_dls_list () in
-    let (key, init) = k in
-    let rec search k l =
+    let rec search key_id init dls_list l =
       match l with
       | [] ->
         begin
           let r = Obj.repr (init ()) in
```

Comment:
You may consider the following refactoring:

```ocaml
let slot = Obj.repr (init ()) in
set_dls_list ({key_id; slot}::dls_list);
slot
```

Reply:
Good suggestion, took it. Thanks!